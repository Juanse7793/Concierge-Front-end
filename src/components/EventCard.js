import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const EventCard = ({ event }) => (
  <Link to={`/events/${event.id}`} className="event-card column">
    <div className="circle">
      <img src={event.image} className="eventImage" alt="" />
    </div>
    <h3 className="event-card-title">{event.name}</h3>
    <p className="event-card-details">{event.location}</p>
    <p className="event-card-details">
      $
      {event.price}
    </p>
  </Link>
);

EventCard.propTypes = {
  event: PropTypes.objectOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
    }),
  ).isRequired,
};

export default EventCard;
