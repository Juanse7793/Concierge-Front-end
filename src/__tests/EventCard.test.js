import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import EventCard from '../components/EventCard';

it('matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <EventCard event={{
        id: 1,
        name: 'Event Name',
        image_urls: ['images/event'],
        location: 'Event Location',
        price: '10',
      }}
      />
      ,
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
