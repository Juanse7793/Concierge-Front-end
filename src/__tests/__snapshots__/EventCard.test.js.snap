// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <a
    className="event-card column"
    href="/events/1"
    onClick={[Function]}
  >
    <div
      className="circle"
    >
      <img
        alt=""
        className="eventImage"
        src="images/event"
      />
    </div>
    <h3
      className="event-card-title"
    >
      Event Name
    </h3>
    <p
      className="event-card-details"
    >
      Event Location
    </p>
    <p
      className="event-card-details"
    >
      $10
    </p>
  </a>,
  ",",
]
`;
